examples/reference/custom_components/ReactComponent.md

```diff
@@ -0,0 +1,429 @@
+       self.value = str(event.__dict__)
+
+button = ButtonEventExample()
+pn.Column(
+    button, pn.widgets.TextAreaInput(value=button.param.value, height=200),
+).servable()
+```
+
+You can also define and send your own custom events:
+
+```python
+import panel as pn
+import param
+
+from panel.custom import ReactComponent
+
+pn.extension()
+
+class CustomEventExample(ReactComponent):
+
+    value = param.Parameter()
+
+    _esm = """
+    function send_event(data) {
+        const currentDate = new Date();
+        const custom_event = new CustomEvent("click", { detail: currentDate.getTime() });
+        data.send_event('click', custom_event)
+    }
+
+    function App(props) {
+        return (
+            <button onClick={e => send_event(props.data)}>
+            Click me
+            </button>
+        );
+    }
+
+    export function render({data}) {
+        return <App data={data}/>;
+    }
+    """
+
+def _handle_click(self, event):
+    self.value = str(event.__dict__)
+
+button = CustomEventExample()
+pn.Column(
+    button, pn.widgets.TextAreaInput(value=button.param.value, height=200),
+).servable()
+```
+
+## Dependency Imports
+
+JavaScript dependencies can be directly imported via URLs, such as those from [`esm.sh`](https://esm.sh/).
+
+```python
+import panel as pn
+
+from panel.custom import ReactComponent
+
+pn.extension()
+
+class ConfettiButton(ReactComponent):
+
+    _esm = """
+    import confetti from "https://esm.sh/canvas-confetti@1.6.0";
+
+    export function render() {
+        return (
+            <button onClick={e => confetti()}>
+            Click Me
+            </button>
+        );
+    }
+    """
+
+ConfettiButton().servable()
+```
+
+Use the `_import_map` attribute for more concise module references.
+
+```python
+import panel as pn
+
+from panel.custom import ReactComponent
+
+pn.extension()
+
+class ConfettiButton(ReactComponent):
+    _importmap = {
+        "imports": {
+            "canvas-confetti": "https://esm.sh/canvas-confetti@1.6.0",
+        }
+    }
+
+    _esm = """
+    import confetti from "canvas-confetti";
+
+    export function render() {
+        return (
+            <button onClick={e => confetti()}>
+            Click Me
+            </button>
+        );
+    }
+    """
+
+ConfettiButton().servable()
+```
+
+See [import map](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap) for more info.
+
+## External Files
+
+You can load JSX and CSS from files by providing the paths to these files.
+
+Create the file **counter_button.py**.
+
+```python
+from pathlib import Path
+
+import param
+import panel as pn
+
+from panel.custom import ReactComponent
+
+pn.extension()
+
+class CounterButton(ReactComponent):
+
+    value = param.Integer()
+
+    _esm = "counter_button.jsx"
+    _stylesheets = [Path("counter_button.css")]
+
+CounterButton().servable()
+```
+
+Now create the file **counter_button.jsx**.
+
+```javascript
+function App(props) {
```

Comment:
Autoreload no longer works for the `ReactComponent` JS/ JSX file.

When I change it the app/ component does not automatically update. I have to click refresh in the browser to see the change.


https://github.com/holoviz/panel/assets/42288570/4a914c29-958f-4836-bc71-b2b4ef317935

It works for the css file. Everything works for the `JSComponent`.